zio-http/src/main/scala/zio/http/codec/SimpleCodec.scala

```diff
@@ -49,4 +49,16 @@ object SimpleCodec {
 
     def isDefinedAt(input: A): Boolean = true
   }
+
+  final case class Custom[Input, Output](f: PartialFunction[Input, Output], g: Output => Input)
```

Comment:
I am not sure this is the right place to solve the problem, or, indeed, that it is necessary to solve it by adding something new (as opposed to re-using alternation).

Basically, can we write the following function:

```scala
def outErrors[Err2](map: Map[Status, Err2])(implicit ...)
```

I think we can write it using only `|` and `transformOrFail`.

(Why `Map` and why not a pair of functions? Because the status code contains no details so every status code can be mapped to something like a case object of a sealed trait; there's no extra content to include in the error.)

I also have my suspicious that what is wanted here is not so much "multiple status codes", but multiple error types.

In which case one could imagine something like:

```scala
def outErrors[Sum, Term1 <: Sum: Tag, Term2 <: Sum: Tag](
  term1: HttpCodec[CodecType.Status & CodecType.Content, Term1],
  term2: ...
)
```

So you do something like:

```scala
endpoint.outErrors(error1, error2, error3)
```

where:

```scala
val error1 = StatusCodec.status[MyError1](Status.NotFound)
```

where `status` takes an implicit `Schema[MyError1]` and uses that to define the body.

Reply:
I first wanted to do something similar to this but I was thinking of somehow automatically combining multiple `outError`s having the same supertype into the supertype instead of an `Either`. Then chose to try to solve it differently. 

Your proposed version definitely looks much cleaner. Although the `SimpleCodec` with being a partial function already, seemed like something that was inteded to support this.

Reply:
Maybe we could do with: `HttpCodec.enum(..., ..., ...)` just as a way to simplify the subtyping stuff.

Although used for errors, it could be used for other things too.